shotover-proxy/examples/windsock/cassandra_protocols.rs

```diff
@@ -0,0 +1,210 @@
+use super::cassandra::*;
+use async_trait::async_trait;
+use cdrs_tokio::cluster::session::{SessionBuilder, TcpSessionBuilder};
+use cdrs_tokio::cluster::NodeTcpConfigBuilder;
+use cdrs_tokio::frame::Version;
+use cdrs_tokio::load_balancing::RoundRobinLoadBalancingStrategy;
+use cdrs_tokio::query_values;
+use std::sync::Arc;
+use std::{
+    collections::HashMap,
+    time::{Duration, Instant},
+};
+use test_helpers::{
+    docker_compose::docker_compose, flamegraph::Perf, shotover_process::ShotoverProcessBuilder,
+};
+use tokio::sync::mpsc::UnboundedSender;
+use windsock::{Bench, Report};
+
+pub enum CassandraProtocol {
+    V3,
+    V4,
+    V5,
+}
+
+pub struct CassandraProtocolBench {
+    shotover: Shotover,
+    protocol: CassandraProtocol,
+    topology: Topology,
+    // TODO compression
+}
+
+impl CassandraProtocolBench {
+    pub fn new(protocol: CassandraProtocol, topology: Topology, shotover: Shotover) -> Self {
+        CassandraProtocolBench {
+            protocol,
+            topology,
+            shotover,
+        }
+    }
+}
+
+#[async_trait]
+impl Bench for CassandraProtocolBench {
+    fn tags(&self) -> HashMap<String, String> {
+        [
+            ("name".to_owned(), "cassandra_protocol".to_owned()),
+            (
+                "protocol".to_owned(),
+                match self.protocol {
+                    CassandraProtocol::V3 => "v3".to_owned(),
+                    CassandraProtocol::V4 => "v4".to_owned(),
+                    CassandraProtocol::V5 => "v5".to_owned(),
+                },
+            ),
+            (
+                "topology".to_owned(),
+                match self.topology {
+                    Topology::Single => "single".to_owned(),
+                    Topology::Cluster3 => "cluster3".to_owned(),
+                },
+            ),
+            (
+                "shotover".to_owned(),
+                match self.shotover {
+                    Shotover::None => "none".to_owned(),
+                    Shotover::Standard => "standard".to_owned(),
+                    _ => todo!(),
+                },
+            ),
+        ]
+        .iter()
+        .cloned()
+        .collect()
+    }
+
+    async fn run(
+        &self,
+        flamegraph: bool,
+        _local: bool,
+        _runtime_seconds: u32,
```

Comment:
same as https://github.com/shotover/shotover-proxy/pull/1170/files#r1203455859
we also want to be handling `operations-per-second` arg which the BenchTask may be able to help with (or maybe BenchTask abstractions doesnt work with cdrs-tokio? lets discuss.)